BurraAbhishek/SpeechVision
# Add the trained model

This model is based on the COCO Dataset and is available on the TensorFlow Lite library



## What is the purpose of this pull request? 
This pull request is to upload the trained datasets which can be used for object classification
EDIT: The supporting DART files are also included in the lib directory

## Describe the proposed changes: 
Adding the models in the assets folder loads the model as a saved tflite model. When the application is installed, it saves the model with the relevant metadata, including the labels.txt file.
To extract labels from any tflite file, convert the tflite file into a zip file and extract the labels from the file.
Adding the supporting dart files which create the UI and runs this application

## How to test those changes? 
In my case, I used Android Studio to test the changes, although IntelliJ IDEA can also be used. The steps are as follows:
1. If you haven't cloned the repository yet, open Android Studio, go to New -> Project from Version Control -> Git, and type `https://github.com/BurraAbhishek/SpeechVision.git` in the URL..
2. Install Flutter and Dart plugins, if you haven't already done so.
3. Either use a physical device or an Android Emulator. In my test case, I used Android Q (API 29).
4. Select Android SDK for x86, select your device, and run the program.
5. While the program is running, select 'Take Photo'.
6. Capture a photograph and see the results.
